ui/src/user/containers/__tests__/SignUpPageContainer.test.jsx

```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import { mount } from 'enzyme';
+import { Provider } from 'react-redux';
+
+import { getStore } from '../../../fixtures/store';
+import { userSignUp } from '../../../actions/user';
+import SignUpPageContainer from '../SignUpPageContainer';
+import SignUpPage from '../../components/SignUpPage';
+
+jest.mock('../../../actions/user');
+userSignUp.mockReturnValue(async () => {});
+
+describe('SignUpPageContainer', () => {
+  it('calls userSignUp onLoginClick', () => {
+    const store = getStore();
+    const wrapper = mount(
+      <Provider store={store}>
+        <SignUpPageContainer />
+      </Provider>
+    );
+    const onSignUpClick = wrapper.find(SignUpPage).prop('onSignUpClick');
+    const dataToPass = {
+      email: 'jessica@jones.com',
+    };
+    onSignUpClick(dataToPass);
+    expect(userSignUp).toHaveBeenCalledWith(dataToPass);
```

Comment:
Lately we started dropping this approach to test if container is dispatching the right action or not. Because this will pass even if you do `userSignUp(data)` without `dispatch(userSignUp(data))`.

Instead you can add `USER_SIGN_UP_REQUEST` (would be useful only for debugging and testing), and assert this here via `store.getActions`. I know that that goes further by testing the action creator's logic (wrong), but I think it's still better than this one.

Actually bonus point, you can set `isSigningUp` in the state and manage it there instead of setting `loading: true` in the form `onSubmit` and never changing it back to `false`. Which would cause problems it `userSignUp` ends up throwing an error, user would stay in the `SignUpPage` look at the `loading` button, without any feedback :/